tests/compilerunits/confread/tcli_parsing.nim

```diff
@@ -0,0 +1,81 @@
+discard """
+description: ""
+joinable: false
+"""
+
+import
+  ast/[
+    reports
+  ],
+  front/[
+    options,
+    commands
+  ],
+  std/[
+    unittest,
+    strtabs,
+    strutils,
+    os,
+    algorithm,
+    sequtils
+  ]
+
+
+var reported: seq[Report]
+
+proc hook(conf: ConfigRef, report: Report): TErrorHandling =
+  reported.add report
+  return doNothing
+
+proc getReports(): seq[Report] =
+  result = reported
+  reported = @[]
+
+
+proc parse(cmd: string, pass: TCmdLinePass = passCmd1): tuple[reports: seq[Report], conf: CurrentConf] =
+  var tmp = newConfigRef(hook)
+  processCmdLine(pass, cmd, tmp)
+  result.reports = getReports()
+  result.conf = tmp.active
+
+proc parse(cmds: seq[string], pass: TCmdLinePass = passCmd1): auto =
+  parse(cmds.join(" "), pass)
+
+suite "Basic command parsing":
+  test "Valid commands":
+    let conf = parse("compile file.nim").conf
+
+    check conf.projectName == "file.nim"
+    check conf.cmd == cmdCompileToC
+
+  test "Backend resetting":
+    let conf = parse("c --backend:cpp --backend:js file.nim").conf
+    check conf.projectName == "file.nim"
+    check conf.backend == backendJs
+
+  test "Hint configuration":
+    let conf = parse("--hint=all:off --hint=Processing:on").conf
+    check(conf.noteSets[cnCurrent] * repHintKinds == {rsemProcessing})
+
+  test "Warning configuration":
+    let conf = parse("--warning=all:off --warning=UnusedImport:on").conf
+    check(conf.noteSets[cnCurrent] * repWarningKinds == {rsemUnusedImport})
+
+# import hmisc/other/hpprint
+
+let csd = currentSourcePath().parentDir()
+
+suite "Path options specification":
```

Comment:
Anecdote/Tangent:

This sort of testing is an important step forward, pretty early when I started working on `nimsuggest` I was told about `findNimProjectFile` or some proc like that. Anyhow, the intention was detecting a project file so the vscode extension can infer things in a better way than having to specify a config.

Long story short, `findNimProjectFile` is busted and the whole thing is buggy as well. It really needs a spec and this test case shows a pretty good way of how I can now enumerate that.

So at the time I didn't have a great way of testing it and no one was a fan of adding these sort of tests to the compiler so I ended up dropping it. This PR reminded me of that, so now at least there is a way forward. 🎉